lib/tasks/data_maintenance.rake

```diff
@@ -1,6 +1,17 @@
 # frozen_string_literal: true
 
 namespace :data_maintenance do
+  desc 'remove duplicated moves'
+  task remove_duplicate_moves: :environment do
+    duplicates = Move.select(:nomis_event_ids).group(:nomis_event_ids).having('COUNT(*) > 1').size
+    duplicates.each_pair do |k, _|
+      next if k.nil?
+
+      moves = Move.order(:created_at).where(nomis_event_ids: k).offset(1)
+      moves.destroy_all
+    end
+  end
+
```

Comment:
Hi @ngw, could we add a test for this please? I'm probably reading this wrong, but it looks like `duplicates` is set to be a number (the result of `.size`).

Could we also add some comments, and name variables more explicitly. It's not clear to me what `k` represents, and why it's pulled out with `each_pair`.

Reply:
Morning :Wave:. I've dug into this code a bit more and I think the behaviour is correct. I'd recommend refactoring it to make it more readable, for example:

```
    duplicate_event_ids = Move
                            .group(:nomis_event_ids)
                            .having('COUNT(*) > 1')
                            .pluck(:nomis_event_ids)
                            .compact
    
    duplicate_event_ids.each  do |event_ids|
      Move
        .where(nomis_event_ids: event_ids)
        .order(:created_at)
        .offset(1)
        .destroy_all
    end
```

I also think this needs a test, as it destroys data.